pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py

```diff
@@ -11,85 +11,90 @@
+        "on_train_epoch_start": _LogOptions(on_step=(False, True), on_epoch=(True,)),
+        "on_train_epoch_end": _LogOptions(on_step=(False,), on_epoch=(True,)),
+        "on_validation_epoch_start": _LogOptions(on_step=(False, True), on_epoch=(True,)),
+        "on_validation_epoch_end": _LogOptions(on_step=(False,), on_epoch=(True,)),
+        "on_test_epoch_start": _LogOptions(on_step=(False, True), on_epoch=(True,)),
+        "on_test_epoch_end": _LogOptions(on_step=(False,), on_epoch=(True,)),
+        "on_predict_epoch_start": None,
+        "on_predict_epoch_end": None,
+        "on_epoch_start": _LogOptions(on_step=(False, True), on_epoch=(True,)),
+        "on_epoch_end": _LogOptions(on_step=(False,), on_epoch=(True,)),
+        "on_batch_start": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "on_batch_end": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "on_train_batch_start": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "on_train_batch_end": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "on_validation_batch_start": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "on_validation_batch_end": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "on_test_batch_start": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "on_test_batch_end": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "on_predict_batch_start": None,
+        "on_predict_batch_end": None,
+        "on_keyboard_interrupt": None,
+        "on_exception": None,
+        "on_save_checkpoint": None,
+        "on_load_checkpoint": None,
+        "setup": None,
+        "teardown": None,
+        "configure_sharded_model": None,
+        "training_step": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "validation_step": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "test_step": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "predict_step": None,
+        "training_step_end": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "validation_step_end": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "test_step_end": _LogOptions(on_step=(False, True), on_epoch=(False, True)),
+        "training_epoch_end": _LogOptions(on_step=(False,), on_epoch=(True,)),
+        "validation_epoch_end": _LogOptions(on_step=(False,), on_epoch=(True,)),
+        "test_epoch_end": _LogOptions(on_step=(False,), on_epoch=(True,)),
+        "configure_optimizers": None,
+        "on_train_dataloader": None,
+        "train_dataloader": None,
+        "on_val_dataloader": None,
+        "val_dataloader": None,
+        "on_test_dataloader": None,
+        "test_dataloader": None,
+        "prepare_data": None,
+        "configure_callbacks": None,
+        "on_validation_model_eval": None,
+        "on_test_model_eval": None,
+    }
```

Comment:
as we're refactoring this, does `check_logging` need to be a class method vs a static method?

Reply:
Good point, let's drop _FxValidator in favour of using `_FN_LOG_OPTIONS` = {...} and use `check_logging` as a functional checker.

Reply:
So to be clear, you want this?

```python
# Copyright The PyTorch Lightning team.
_FN_LOG_OPTIONS = {
    ...
}

def _check_logging(fx_name, ...):
    if fx_name not in _FN_LOG_OPTIONS:
        ...
```

I used a class to group things together and in case we needed more validation functions in the future

Reply:
All good !